BLD: Fix x86-simd-sort build failure on openBSD

charris opened the pull request:
Backport of #26797.

Fixes building x86-simd-sort on openBSD. See intel/x86-simd-sort#157 for details.

The following are updated from main

- highway submodule
- vendored-meson submodule 
- x86_simd_sort submodule
- highway-qsort.dispatch.cpp.
- x86_simd_argsort.dispatch.cpp
- x86_simd_qsort.dispatch.cpp
- x86_simd_qsort_16bit.dispatch.cpp

Small fixes are taken from

- numpy/_core/meson.build
 


<!--         ----------------------------------------------------------------
                MAKE SURE YOUR PR GETS THE ATTENTION IT DESERVES!
                ----------------------------------------------------------------

*  FORMAT IT RIGHT:
      https://www.numpy.org/devdocs/dev/development_workflow.html#writing-the-commit-message

*  IF IT'S A NEW FEATURE OR API CHANGE, TEST THE WATERS:
      https://www.numpy.org/devdocs/dev/development_workflow.html#get-the-mailing-list-s-opinion

*  HIT ALL THE GUIDELINES:
      https://numpy.org/devdocs/dev/index.html#guidelines

*  WHAT TO DO IF WE HAVEN'T GOTTEN BACK TO YOU:
      https://www.numpy.org/devdocs/dev/development_workflow.html#getting-your-pr-reviewed
-->

Commit:
BLD: Fix x86-simd-sort build failure on openBSD

charris committed:
MAINT: Update highway to 1dbb1180.

charris commented:
@rgommers Should we update `vendored-meson` as well. I've been conservative about that.

charris commented:
@r-devulap I assume that there is a missing backport here. Do you know what it is?

r-devulap commented:
probably this one? https://github.com/intel/x86-simd-sort/pull/143

rgommers commented:
> @rgommers Should we update `vendored-meson` as well. I've been conservative about that.

Working on it at https://github.com/numpy/meson/pull/15

charris committed:
MAINT: Update various sort files from main

- Update highway-qsort.dispatch.cpp.
- Update x86_simd_argsort.dispatch.cpp
- Update x86_simd_qsort.dispatch.cpp
- Update x86_simd_qsort_16bit.dispatch.cpp

charris commented:
> Working on it at 

I suspect the new risc failure may be meson connected.

charris committed:
MAINT: Update vendored-meson from main

mattip commented:
There is a failure on the riscv qemu build @r-devulap does this ring a bell?
```
FAILED: numpy/_core/libhighway.a.p/src_highway_hwy_targets.cc.o 
c++ -Inumpy/_core/libhighway.a.p -Inumpy/_core -I../numpy/_core -I../numpy/_core/src/highway -I/numpy/build/meson_cpu -fdiagnostics-color=always -D_GLIBCXX_ASSERTIONS=1 -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -std=c++17 -O2 -g -fPIC -DTOOLCHAIN_MISS_ASM_HWCAP_H -MD -MQ numpy/_core/libhighway.a.p/src_highway_hwy_targets.cc.o -MF numpy/_core/libhighway.a.p/src_highway_hwy_targets.cc.o.d -o numpy/_core/libhighway.a.p/src_highway_hwy_targets.cc.o -c ../numpy/_core/src/highway/hwy/targets.cc
../numpy/_core/src/highway/hwy/targets.cc: Assembler messages:
../numpy/_core/src/highway/hwy/targets.cc:631: Error: unrecognized opcode `vsetvli s0,zero,e8,m1,ta,ma'

```

r-devulap commented:
@mattip I think you need to backport this PR https://github.com/numpy/numpy/pull/26273 as well.

charris commented:
> I think you need to backport this PR #26273 as well. 

Looking at the diff of `numpy/_core/meson.build`, there seem to be a few other missing pieces as well. Is the problem fixed by #26797 also present in  2.0.0? If not, I'm inclined to skip this backport, it is getting overly intrusive.

r-devulap commented:
> Looking at the diff of `numpy/_core/meson.build`, there seem to be a few other missing pieces as well. Is the problem fixed by #26797 also present in 2.0.0? If not, I'm inclined to skip this backport, it is getting overly intrusive.

unfortunately, yes :( The original [bug report](https://github.com/intel/x86-simd-sort/issues/156) was numpy 2.0.0

charris committed:
MAINT: Update "numpy/_core/meson.build" from #26273
